core/src/main/java/org/apache/iceberg/CatalogProperties.java

```diff
@@ -30,6 +30,21 @@ private CatalogProperties() {
   public static final String FILE_IO_IMPL = "io-impl";
   public static final String WAREHOUSE_LOCATION = "warehouse";
 
+  public static final String CACHE_ENABLED = "cache-enabled";
```

Comment:
nit: I talked about just removing it, but I am also fine if we just keep it.

In that case, could you add a javadoc about it?

Also with this change we also need to deprecate `FlinkCatalogFactory.CACHE_ENABLED`

Reply:
I was going to handle Flink in a separate PR (as Flink uses a separate variable for its `cache-enabled` anyway). I'll add a deprecation notice to it though.

I'll add a javadoc comment here as well.